utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -794,3 +765,42 @@ impl From<DisabledSyscallsInvocator> for ModuleWithCallIndexes {
         }
     }
 }
+
+#[derive(Clone, Debug)]
+struct ParamInstructions(Vec<Instruction>);
+
+impl ParamInstructions {
+    fn into_inner(self) -> Vec<Instruction> {
+        self.0
+    }
+
+    fn as_i32(&self) -> Option<i32> {
+        if self.0.len() != 1 {
+            return None;
+        }
+
+        if let Some(Instruction::I32Const(ret)) = self.0.last() {
+            Some(*ret)
+        } else {
+            None
+        }
+    }
+}
+
+impl From<i32> for ParamInstructions {
+    fn from(value: i32) -> Self {
+        ParamInstructions(vec![Instruction::I32Const(value)])
+    }
+}
+
+impl From<i64> for ParamInstructions {
+    fn from(value: i64) -> Self {
+        ParamInstructions(vec![Instruction::I64Const(value)])
+    }
+}
```

Comment:
I think it's better to do explicitly

Reply:
Definitely no. It was very-very noisy without it.